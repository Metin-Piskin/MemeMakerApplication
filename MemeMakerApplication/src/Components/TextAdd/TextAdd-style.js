import { StyleSheet } from "react-native";

export default StyleSheet.create({
    imput: {
        backgroundColor: '#fff',
        margin: 0,
        borderRadius: 10,
        padding: 7,
        flex: 1
    },
    imputnull: {
        backgroundColor: '#2C2C2E',
        margin: 0,
        borderRadius: 10,
        padding: 7,
        flex: 1
    },
    addtextbutton: {
        backgroundColor: '#161718',
        alignItems: 'center',
        justifyContent: 'center',
        paddingHorizontal: 10,
        marginLeft: 10,
        borderRadius: 10,
        borderWidth: 4,
        borderColor: '#fff',
    },
    addtextbuttonnull: {
        backgroundColor: '#161718',
        alignItems: 'center',
        justifyContent: 'center',
        paddingHorizontal: 10,
        marginLeft: 10,
        borderRadius: 10,
        borderWidth: 4,
        borderColor: '#2C2C2E',
    },
    addtext: {
        color: '#fff',
        fontFamily: 'SpecialElite-Regular',
    },
    addtextnull: {
        color: '#2C2C2E',
        fontFamily: 'SpecialElite-Regular',
    },
    editcontainer: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        paddingVertical: 10,
        paddingHorizontal: 15
    },
})