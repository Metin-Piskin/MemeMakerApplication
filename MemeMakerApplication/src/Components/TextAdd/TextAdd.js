import React from 'react';
import { View, Text, TextInput, TouchableOpacity } from 'react-native';

import styles from './TextAdd-style';

const TextAdd = ({
    btn,
    placeholder,
    onChangeText,
    placeholderTextColor,
    multiline,
    onPress,
    disabled,
}) => {
    return (
        <View style={styles.editcontainer} testID='text-add-container'>
            {
                btn ? (
                    <>
                        <TextInput
                            testID='text-add-input-null'
                            placeholder={placeholder}
                            onChangeText={onChangeText}
                            style={styles.imputnull}
                            placeholderTextColor={placeholderTextColor}
                            multiline={multiline}
                        />
                        <TouchableOpacity
                            onPress={onPress}
                            testID='text-add-press-null'
                            disabled={disabled}
                            style={styles.addtextbuttonnull}
                        >
                            <Text style={styles.addtextnull} testID='text-add-press-text-null'>
                                ADD
                            </Text>
                        </TouchableOpacity>
                    </>
                ) : (
                    <>
                        <TextInput
                            testID='text-add-input'
                            placeholder={placeholder}
                            onChangeText={onChangeText}
                            style={styles.imput}
                            placeholderTextColor={placeholderTextColor}
                            multiline={multiline}
                        />
                        <TouchableOpacity
                            onPress={onPress}
                            testID='text-add-press'
                            disabled={disabled}
                            style={styles.addtextbutton}
                        >
                            <Text style={styles.addtext} testID='text-add-press-text'>
                                ADD
                            </Text>
                        </TouchableOpacity>
                    </>
                )
            }

        </View>

    )
}

export default TextAdd;