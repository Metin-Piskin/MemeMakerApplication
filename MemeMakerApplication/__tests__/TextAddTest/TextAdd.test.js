import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';

import TextAdd from '../../src/Components/TextAdd';
import TextAddStyles from '../../src/Components/TextAdd/TextAdd-style';

test('renders correctly', () => {
    const comp = render(<TextAdd />);
    expect(comp).toMatchSnapshot();
});

test('renders correctly', () => {
    const test = true
    const comp = render(<TextAdd btn={test} />);
    expect(comp).toMatchSnapshot();
});

test('should trigerr onPress', () => {
    const mockFunction = jest.fn();
    const comp = render(<TextAdd onPress={mockFunction} />);

    const ButtonTouchable = comp.getByTestId('text-add-press');
    fireEvent(ButtonTouchable, 'press');

    expect(mockFunction).toBeCalled();
})

test('should render default theme style', () => {
    const comp = render(<TextAdd />);

    const TextAddContainerStyle = comp.getByTestId('text-add-container').props.style;

    expect(TextAddContainerStyle).toMatchObject(TextAddStyles.editcontainer);
});

test('should render default theme style', () => {
    const comp = render(<TextAdd />);

    const TextAddImputStyle = comp.getByTestId('text-add-input').props.style;

    expect(TextAddImputStyle).toMatchObject(TextAddStyles.imput);
});

test('should render default theme style', () => {
    const comp = render(<TextAdd />);

    const TextAddPressStyle = comp.getByTestId('text-add-press').props.style;

    expect(TextAddPressStyle).toMatchObject(TextAddStyles.addtextbutton);
});

test('should render default theme style', () => {
    const comp = render(<TextAdd />);

    const TextAddPressTextStyle = comp.getByTestId('text-add-press-text').props.style;

    expect(TextAddPressTextStyle).toMatchObject(TextAddStyles.addtext);
});

test('should render default theme style', () => {
    const test = true
    const comp = render(<TextAdd btn={test} />);

    const TextAddImputNullStyle = comp.getByTestId('text-add-input-null').props.style;

    expect(TextAddImputNullStyle).toMatchObject(TextAddStyles.imputnull);
});

test('should render default theme style', () => {
    const test = true
    const comp = render(<TextAdd btn={test} />);

    const TextAddPressNullStyle = comp.getByTestId('text-add-press-null').props.style;

    expect(TextAddPressNullStyle).toMatchObject(TextAddStyles.addtextbuttonnull);
});

test('should render default theme style', () => {
    const test = true
    const comp = render(<TextAdd btn={test} />);

    const TextAddPressTextNullStyle = comp.getByTestId('text-add-press-text-null').props.style;

    expect(TextAddPressTextNullStyle).toMatchObject(TextAddStyles.addtextnull);
});