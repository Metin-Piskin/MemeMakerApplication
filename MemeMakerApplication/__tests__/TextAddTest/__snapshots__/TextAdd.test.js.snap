// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 15,
      "paddingVertical": 10,
    }
  }
  testID="text-add-container"
>
  <TextInput
    style={
      {
        "backgroundColor": "#fff",
        "borderRadius": 10,
        "flex": 1,
        "margin": 0,
        "padding": 7,
      }
    }
    testID="text-add-input"
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#161718",
        "borderColor": "#fff",
        "borderRadius": 10,
        "borderWidth": 4,
        "justifyContent": "center",
        "marginLeft": 10,
        "opacity": 1,
        "paddingHorizontal": 10,
      }
    }
    testID="text-add-press"
  >
    <Text
      style={
        {
          "color": "#fff",
          "fontFamily": "SpecialElite-Regular",
        }
      }
      testID="text-add-press-text"
    >
      ADD
    </Text>
  </View>
</View>
`;

exports[`renders correctly 2`] = `
<View
  style={
    {
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 15,
      "paddingVertical": 10,
    }
  }
  testID="text-add-container"
>
  <TextInput
    style={
      {
        "backgroundColor": "#2C2C2E",
        "borderRadius": 10,
        "flex": 1,
        "margin": 0,
        "padding": 7,
      }
    }
    testID="text-add-input-null"
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#161718",
        "borderColor": "#2C2C2E",
        "borderRadius": 10,
        "borderWidth": 4,
        "justifyContent": "center",
        "marginLeft": 10,
        "opacity": 1,
        "paddingHorizontal": 10,
      }
    }
    testID="text-add-press-null"
  >
    <Text
      style={
        {
          "color": "#2C2C2E",
          "fontFamily": "SpecialElite-Regular",
        }
      }
      testID="text-add-press-text-null"
    >
      ADD
    </Text>
  </View>
</View>
`;
